x-pack/agent/pkg/reporter/fleet/reporter.go

```diff
@@ -41,136 +40,111 @@ func (e *event) Message() string {
 	return e.Msg
 }
 
-type checkinExecutor interface {
-	Execute(r *fleetapi.CheckinRequest) (*fleetapi.CheckinResponse, error)
-}
-
-type remoteClient interface {
-	Send(
-		method string,
-		path string,
-		params url.Values,
-		headers http.Header,
-		body io.Reader,
-	) (*http.Response, error)
-}
-
 // Reporter is a reporter without any effects, serves just as a showcase for further implementations.
 type Reporter struct {
-	logger         *logger.Logger
-	queue          []reporter.Event
-	qlock          sync.Mutex
-	ticker         *time.Ticker
-	threshold      int
-	droppedCounter int
-	checkingCmd    checkinExecutor
-	closeChan      chan struct{}
-	closeOnce      sync.Once
+	logger    *logger.Logger
+	queue     []fleetapi.SerializableEvent
+	qlock     sync.Mutex
+	threshold int
+	lastAck   time.Time
 }
 
 type agentInfo interface {
 	AgentID() string
 }
 
 // NewReporter creates a new fleet reporter.
-func NewReporter(agentInfo agentInfo, l *logger.Logger, c *ManagementConfig, client remoteClient) (*Reporter, error) {
-	checkinClient := fleetapi.NewCheckinCmd(agentInfo, client)
-
-	frequency := time.Duration(c.ReportingCheckFrequency) * time.Second
+func NewReporter(agentInfo agentInfo, l *logger.Logger, c *ManagementConfig) (*Reporter, error) {
 	r := &Reporter{
-		queue:       make([]reporter.Event, 0),
-		ticker:      time.NewTicker(frequency),
-		logger:      l,
-		checkingCmd: checkinClient,
-		threshold:   c.Threshold,
-		closeChan:   make(chan struct{}),
+		queue:     make([]fleetapi.SerializableEvent, 0),
+		logger:    l,
+		threshold: c.Threshold,
 	}
 
-	go r.reportLoop()
 	return r, nil
 }
 
-// Report in noop reporter does nothing.
+// Report enqueue event into reporter queue.
 func (r *Reporter) Report(e reporter.Event) error {
 	r.qlock.Lock()
 	defer r.qlock.Unlock()
 
-	r.queue = append(r.queue, e)
+	r.queue = append(r.queue, &event{
+		EventType: e.Type(),
+		Ts:        fleetapi.Time(e.Time()),
+		SubType:   e.SubType(),
+		Msg:       e.Message(),
+		Payload:   e.Payload(),
+		Data:      e.Data(),
+	})
+
 	if r.threshold > 0 && len(r.queue) > r.threshold {
+		// drop some low importance event if needed
 		r.dropEvent()
```

Comment:
I know we will change that logic by adding priority, but I think we should at least produce a log when we drop something.

Reply:
logging in specific drop events so we can include event in a log